service/model/config.go

```diff
@@ -0,0 +1,139 @@
+package model
+
+import (
+	"errors"
+	"fmt"
+)
+
+var (
+	errOrganizationNameRequired        = errors.New("organization name is required")
+	errDefaultReleaseMsgRequired       = errors.New("default release message is required")
+	errIntegrationMissingToken         = errors.New("token is required to enable integration")
+	errIntegrationTokenSetWhenDisabled = errors.New("token cannot be set when integration is disabled")
+)
+
+type Config struct {
+	OrganizationName      string
+	DefaultReleaseMessage string
+	Slack                 IntegrationConfig
+	Github                IntegrationConfig
+}
+
+type ConfigUpdate struct {
+	OrganizationName  *string
+	DefaultReleaseMsg *string
+	Slack             IntegrationConfigUpdate
+	Github            IntegrationConfigUpdate
+}
+
+func (c *Config) Update(u ConfigUpdate) error {
+	if u.OrganizationName != nil {
+		c.OrganizationName = *u.OrganizationName
+	}
+	if u.DefaultReleaseMsg != nil {
+		c.DefaultReleaseMessage = *u.DefaultReleaseMsg
+	}
+
+	if err := c.Slack.Update(u.Slack); err != nil {
+		return fmt.Errorf("slack: %w", err)
+	}
+	if err := c.Github.Update(u.Github); err != nil {
+		return fmt.Errorf("github: %w", err)
+	}
+
+	return c.Validate()
+}
+
+func (c *Config) Validate() error {
+	if c.OrganizationName == "" {
+		return errOrganizationNameRequired
+	}
+
+	if c.DefaultReleaseMessage == "" {
+		return errDefaultReleaseMsgRequired
+	}
+
+	return nil
+}
+
+func ToConfig(
+	orgName,
+	rlsMsg string,
+	slackEnabled bool,
+	slackToken string,
+	githubEnabled bool,
```

Comment:
In general, probably no need to do here, but in general, if the above comment taken to the extreme, you might want to wrap every input string into a custom type, i.e. in this case slackToken will have `SlackToken` type (which will simply embed `string`), etc.

Might sound extreme, but given the arguments above, might be worth it.
I took this idea from a conference from a cool Google guy or similar...

Yes, no need to do it here, because if you agree to wrap it into a type, it already will be much safer...